File: amwell-react-native-demo/react-native-amwellservice/ios/src/AmwellViewManager.m
//
//  AmwellViewController.m
//  Amwellservice
//
//  Created by YUNFENG LIN on 12/24/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

#import <React/RCTViewManager.h>

// the component will be exposed as "CounterView"
// if the name ends with "Manager" it will be stripped by React Native
@interface RCT_EXTERN_MODULE(AmwellViewManager, RCTViewManager)
// or, we could also rename it ourselves
//@interface RCT_EXTERN_REMAP_MODULE(CounterView, CounterViewManager, RCTViewManager)

// all NSNumber arguments are required to be marked as "nonnull"
RCT_EXTERN_METHOD(updateValueViaManager:(nonnull NSNumber *)node)
RCT_EXPORT_VIEW_PROPERTY(onUpdate, RCTDirectEventBlock)

@end

File: amwell-react-native-demo/react-native-amwellservice/ios/src/AmwellViewManager.swift
//
//  AmwellViewController.swift
//  Amwellservice
//
//  Created by YUNFENG LIN on 12/24/19.
//  Copyright © 2019 Facebook. All rights reserved.
//

import Foundation
import AWSDK
import SwiftUI

@objc(AmwellViewManager)
class AmwellViewManager: RCTViewManager {
    
    var viewController : AmwellViewController?
    
    override init() {
        viewController = AmwellViewController()
    }
    
    override func view() -> UIView! {
        return viewController?.view
    }
    
    // this is required since RN 0.49+
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    // MARK: Bridged methods
    @objc func testVideoCall(_ node: NSNumber) {
        DispatchQueue.main.async {
            self.viewController!.demoVideoCall()
        }
    }
}
